File: app/src/main/java/cn/jingzhuan/lib/chart/demo/BarChartModel.java
package cn.jingzhuan.lib.chart.demo;

import android.databinding.ViewDataBinding;
import android.graphics.Color;

import com.airbnb.epoxy.DataBindingEpoxyModel;
import com.airbnb.epoxy.EpoxyModelClass;

import java.util.ArrayList;
import java.util.List;

import cn.jingzhuan.lib.chart.demo.databinding.LayoutBarChartBinding;
import cn.jingzhuan.lib.chart.data.BarDataSet;
import cn.jingzhuan.lib.chart.data.BarValue;

/**
 * Created by Donglua on 17/8/2.
 */
@EpoxyModelClass(layout = R.layout.layout_bar_chart)
public abstract class BarChartModel extends DataBindingEpoxyModel {

    private BarDataSet barDataSet;

    public BarChartModel() {

        List<BarValue> barValueList = new ArrayList<>();

        barValueList.add(new BarValue(11));
        barValueList.add(new BarValue(10));
        barValueList.add(new BarValue(11));
        barValueList.add(new BarValue(13));
        barValueList.add(new BarValue(11));
        barValueList.add(new BarValue(12));
        barValueList.add(new BarValue(12));
        barValueList.add(new BarValue(13, Color.DKGRAY));
        barValueList.add(new BarValue(15));

        barDataSet = new BarDataSet(barValueList);
        barDataSet.setAutoBarWidth(true);
    }

    @Override
    protected void setDataBindingVariables(ViewDataBinding binding) {
        if (binding instanceof LayoutBarChartBinding) {

            LayoutBarChartBinding barBinding = (LayoutBarChartBinding) binding;

            barBinding.barChart.setDataSet(barDataSet);
            barBinding.barChart.getAxisRight().setLabelTextColor(Color.BLACK);
        }
    }
}

File: demo-chart2/src/main/java/cn/jingzhuan/lib/chart2/demo/BarChartModel.kt
package cn.jingzhuan.lib.chart2.demo

import androidx.databinding.ViewDataBinding
import android.graphics.Color
import android.graphics.Paint
import android.view.View
import android.view.ViewGroup
import cn.jingzhuan.lib.chart.data.BarDataSet
import cn.jingzhuan.lib.chart.data.BarValue
import cn.jingzhuan.lib.chart2.demo.databinding.LayoutBarChartBinding
import com.airbnb.epoxy.DataBindingEpoxyModel
import com.airbnb.epoxy.EpoxyModelClass
import java.util.ArrayList

/**
 * Created by Donglua on 17/8/2.
 */
@EpoxyModelClass(layout = R.layout.layout_bar_chart)
abstract class BarChartModel : DataBindingEpoxyModel() {

  private val barDataSet: BarDataSet

  init {

    val barValueList = ArrayList<BarValue>()

    barValueList.add(BarValue(11f))
    barValueList.add(BarValue(10f))
    barValueList.add(BarValue(11f))
    barValueList.add(BarValue(13f))
    barValueList.add(BarValue(11f))
    barValueList.add(BarValue(12f).apply { color = Color.RED })
    barValueList.add(BarValue(12f).apply { color = Color.RED })
    barValueList.add(BarValue(13f).apply { setGradientColors(Color.WHITE, Color.BLACK) })
    barValueList.add(BarValue(15f).apply { setGradientColors(Color.WHITE, Color.BLACK) })
    barValueList.add(BarValue(13f).apply {
      color = Color.RED
      paintStyle = Paint.Style.STROKE
    })
    barValueList.add(BarValue(15f).apply {
      color = Color.GREEN
      paintStyle = Paint.Style.STROKE
    })

    barDataSet = BarDataSet(barValueList)
    barDataSet.isAutoBarWidth = true
    barDataSet.strokeThickness = 3f
  }

  override fun buildView(parent: ViewGroup): View {
    val rootView = super.buildView(parent)

    val barBinding = rootView.tag as LayoutBarChartBinding

    barBinding.barChart.setDataSet(barDataSet)
    barBinding.barChart.axisRight.labelTextColor = Color.BLACK

    return rootView
  }

  override fun setDataBindingVariables(binding: ViewDataBinding) {
    binding as LayoutBarChartBinding

    binding.barChart.animateY(500)
  }
}
